vtex-sites/storecomponents.store
# Add PWA configuration and required plugins

## What's the purpose of this pull request?

This PR turns `storecomponents.store` into a PWA 🎉 .

## How it works? 

Using `gatsby-plugin-offline` and `gatsby-plugin-manifest`, along with a custom bit of JS code, Gatsby adds a manifest to the store and registers a service worker. This service worker handles caching and offline behavior for the store. And with that, it meets the installation criteria for a PWA.

## How to test it?

1. Go to the deploy preview URL.
2. You should see an icon, like this in your browser's address bar (if you're using Chrome):
    ![Screen Shot 2021-04-08 at 10 47 36](https://user-images.githubusercontent.com/27777263/114038004-dd75e580-9857-11eb-802e-359cf9d7dd46.png)
    This icon indicates that the site you're in is installable as a PWA.
3. You can install it into your computer
4. Open dev tools and check how the page behaves when offline.

## References

- https://web.dev/progressive-web-apps/
